openwisp_monitoring/check/classes/iperf.py

```diff
@@ -0,0 +1,186 @@
+            sent_bytes = sent_json['bytes']
+            sent_bytes_GB = sent_bytes / 1000000000
+            sent_bps = sent_json['bits_per_second']
+            sent_Gbps = sent_bps / 1000000000
+            received_bytes = recv_json['bytes']
+            received_bytes_GB = received_bytes / 1000000000
+            received_bps = recv_json['bits_per_second']
+            received_Gbps = received_bps / 1000000000
+            retransmits = sent_json['retransmits']
+
+            result = {
+                'iperf_result': 1,
+                'sent_bps': round(sent_Gbps, 2),
+                'received_bps': round(received_Gbps, 2),
+                'sent_bytes': round(sent_bytes_GB, 2),
+                'received_bytes': round(received_bytes_GB, 2),
+                'retransmits': retransmits,
+            }
+            return result
+        # For UDP
+        elif mode == 'UDP':
+            jitter_ms = res_dict['end']['sum']['jitter_ms']
+            packets = res_dict['end']['sum']['packets']
+            lost_packets = res_dict['end']['sum']['lost_packets']
+            lost_percent = float(res_dict['end']['sum']['lost_percent'])
+            result = {
+                'jitter': round(jitter_ms, 2),
+                'packets': packets,
+                'lost_packets': lost_packets,
+                'lost_percent': round(lost_percent, 2),
+            }
+            return result
+
+    def store_result(self, result):
+        """
+        store result in the DB
+        """
+        metric = self._get_metric()
+        copied = result.copy()
+        iperf_result = copied.pop('iperf_result')
+        metric.write(iperf_result, extra_values=copied)
+
+    def store_result_fail(self):
+        """
+        store fail result in the DB
+        """
+        self.store_result(
+            {
+                'iperf_result': 0,
+                'sent_bps': 0.0,
+                'received_bps': 0.0,
+                'sent_bytes': 0.0,
+                'received_bytes': 0.0,
+                'retransmits': 0,
+                'jitter': 0.0,
+                'packets': 0,
+                'lost_packets': 0,
+                'lost_percent': 0.0,
+            }
+        )
+
+    def _get_metric(self):
+        """
+        Gets or creates metric
+        """
+        metric, created = self._get_or_create_metric()
+        if created:
+            self._create_charts(metric)
+        return metric
+
+    def _create_charts(self, metric):
+        """
+        Creates iperf related charts (Bandwith/Jitter)
+        """
+        charts = [
+            'bitrate',
+            'transfer',
+            'retransmits',
+            'jitter',
+            'datagram',
+            'datagram_loss',
+        ]
+        for chart in charts:
+            chart = Chart(metric=metric, configuration=chart)
+            chart.full_clean()
+            chart.save()
+
+    def _create_alert_settings(self, metric):
+        pass
```

Comment:
What should trigger this alert? Is it going to be something like `if bandwidth falls below X Mbps, then send alert`.

Reply:
We can do that as well, but according to **possible outcomes** of the project ''If an Iperf server is down/busy we should trigger an alert" (We need to investigate more about this)